x/ibc/20-transfer/module.go

```diff
@@ -257,5 +317,19 @@ func (am AppModule) OnTimeoutPacket(
 	if err := types.ModuleCdc.UnmarshalBinaryBare(packet.GetData(), &data); err != nil {
 		return nil, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-20 transfer packet data: %s", err.Error())
 	}
-	return handleTimeoutDataTransfer(ctx, am.keeper, packet, data)
+	// refund tokens
+	if err := am.keeper.OnTimeoutPacket(ctx, packet, data); err != nil {
+		return nil, err
+	}
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, AttributeValueCategory),
```

Comment:
this is only an event for the `transfer` category. We should add 2 more events:

* tokens refunded (amount)
* address refunded

```go
sdk.NewEvent(
	EventTimeout,
	sdk.NewAttribute(sdk.AttributeKeyModule, AttributeValueCategory),
        sdk.NewAttribute(AttributeKeyRefundAddress, <address>), // address refunded
        sdk.NewAttribute(AttributeKeyRefundValue, <amount>), // tokens refunded to address above
)